src/components/layout/ChatArea.tsx

```diff
@@ -345,16 +363,41 @@ export const ChatArea: React.FC<{
             `PRIVMSG ${selectedChannel ? selectedChannel.name : ""} :\u0001ACTION ${actionMessage}\u0001`,
           );
         } else {
-          // Handle other commands
           ircClient.sendRaw(
             selectedServerId,
             `${commandName} :${args.join(" ")}`,
           );
         }
       } else {
+        const colorCode = getColorCode(selectedColor || "inherit"); // Get the IRC color code
+
+        // Apply formatting codes
+        let formattedText = messageText;
+        if (selectedFormatting.includes("bold")) {
+          formattedText = `\x02${formattedText}\x02`;
+        }
+        if (selectedFormatting.includes("italic")) {
+          formattedText = `\x1D${formattedText}\x1D`;
+        }
+        if (selectedFormatting.includes("underline")) {
+          formattedText = `\x1F${formattedText}\x1F`;
+        }
+        if (selectedFormatting.includes("strikethrough")) {
+          formattedText = `\x1E${formattedText}\x1E`;
+        }
+        if (selectedFormatting.includes("reverse")) {
+          formattedText = `\x16${formattedText}\x16`;
+        }
+        if (selectedFormatting.includes("monospace")) {
+          formattedText = `\x11${formattedText}\x11`;
+        }
+
+        // Prepend the color code
+        formattedText = `${colorCode}${formattedText}`;
+
```

Comment:
This feels like should be also part of ircUtils or ircFormatting just like colors are :D

Reply:
I feel like it might belong better in the ColorPicker component? what do you think
either way, something for a future "text-formatting" PR, I think

Reply:
not a component, just a .ts for logic maybe